doc/developer/design/20230225_compute_introspection_schema.md

```diff
@@ -0,0 +1,212 @@
+# Compute Introspection sources/tables/views improvements
+
+## Summary
+
+We have identified several issues with the sources documented
+[here](https://materialize.com/docs/sql/system-catalog/mz_internal/). In
+particular, some key views are missing (e.g., a way to connect channel
+IDs to the operators they connect), and there are various existing ones that
+have inconsistently named fields or inconsistent
+semantics. Furthermore, the documentation of the various relations is
+incomplete (e.g., it doesn't make clear wheter each relation is
+per-replica or global). 
+
+## Goals
+
+Compute introspection relations should be useful and easy to understand
+for internal debugging 
+
+## Non-Goals
+
+* The relations do not need to be polished enough for external
+consumption (although it's hoped they prove useful, we don't consider
+it a bug if they're not). Customer-friendly introspection will be
+provided in the web UI by the "external introspection" project.
+
+* We make no attempt to maintain backwards compatibility with
+  previously-existing queries, although as part of this work we will
+  update the internal tools that use the introspection relations.
+
+## Description
+
+### Schema inconsistencies and inelegancies 
+
+In this section, we describe the various issues that make the schema
+confusing.
+
+#### Worker-local vs. global
+
+Several relations have a `worker_id` column. Technically, the
+semantics of this column are the same across all the relations: it
+corresponds to the worker on which the data was measured. However, in
+practice the behavior is very different, and falls into a few separate
+categories, as described below.
+
+##### Sharded relations
+
+*Example:* `mz_arrangement_sizes`. These are relations that report
+some value that is expected to differ across all workers, such that
+the semantically meaningful value is (often) the sum of all the
+per-worker values. For example, a user interested in the total size of
+each arrangement would normally want to issue a query like `SELECT
+operator_id, sum(records), sum(batches) FROM mz_arrangement_sizes
+GROUP BY worker_id`.
+
+There are also a few relations where `sum` is not the most interesting
+aggregate; for example, the `min` across workers in
+`mz_compute_frontiers`. "Sharded" might not be the most precise way to
+describe these, but I include them in this section because the point
+is basically the same.
+
+The existence of the per-worker underlying sources is not completely
+useless, e.g. because they are useful for detecting skew. However,
+they are probably less commonly useful than a summed view would
+be. Unforunately, this summed view does not exist for most such
+relations (one counterexample is `mz_records_per_dataflow` and
+`mz_records_per_dataflow_global`).
```

Comment:
Yes, however this will be fixed by the other suggestions in this document (renaming `mz_records_per_dataflow_global` to `mz_records_per_dataflow` and `mz_records_per_dataflow` to `mz_records_per_dataflow_per_worker`)